osu.Game/Screens/OnlinePlay/Lounge/LoungeSubScreen.cs

```diff
@@ -180,8 +186,9 @@ protected override void LoadComplete()
             searchTextBox.Current.BindValueChanged(_ => updateFilterDebounced());
             ruleset.BindValueChanged(_ => UpdateFilter());
 
-            initialRoomsReceived.BindTo(RoomManager.InitialRoomsReceived);
-            initialRoomsReceived.BindValueChanged(_ => updateLoadingLayer());
+            ListingPollingComponent.InitialRoomsReceived.BindValueChanged(_ => updateLoadingLayer());
```

Comment:
It looks like this is safe due to the internal `InitialRoomsReceived` only being set to true in the scheduled api response, but still reads a bit weird.

```suggestion
            ListingPollingComponent.InitialRoomsReceived.BindValueChanged(received => 
            {
                if (received.NewValue) updateLoadingLayer();
            }, true);
```

I think this would read better.

Reply:
Doesn't the inner condition mean that the loading layer isn't displayed on a filter change?

Reply:
Ah right, I thought it was just hiding. Seems fine as is then. I'll just add the initial invocation then?

Reply:
Sounds good